modepy/shapes.py

```diff
@@ -559,4 +566,40 @@ def _submes_for_hypercube(shape: Hypercube, node_tuples):
 # }}}
 
 
+# {{{ tensor product shape
+
+@dataclass(frozen=True, init=False)
```

Comment:
Is the plan to replace `Hypercube` with this? (I think we should!)

Reply:
We need a `Line` shape then, right? To use as a building block for the hypercube.

Reply:
I'd say `Simplex(1)`.

Reply:
Modified this somewhere in 05fe4d1.

Basically the way things work with tensor products with right now has some `__new__` hackery in it:
* if the tensor product just gets one shape, it skips and just returns that shape
* for `Hypercube`, this means that in 1D it just returns `Simplex(1)`, i.e. there's no `Hypercube(1)`.
* the `TensorProductSpace` and `QN` work exactly the same.